src/services/project/project-router.ts

```diff
@@ -0,0 +1,170 @@
+    ProjectsSchema,
+    CreateProjectRequestSchema,
+    UserAlreadyHasTeamError,
+    UserAlreadyHasTeamErrorSchema,
+    //ProjectMappingSchema,
+    //PathTypeSchema,
+    //TrackTypeSchema
+} from "./project-schema";
+
+//import { EventIdSchema, SuccessResponseSchema } from "../../common/schemas";
+import { z } from "zod";
+import Models from "../../common/models";
+import { getAuthenticatedUser } from "../../common/auth";
+import { UserIdSchema } from "../../common/schemas";
+//import Config from "../../common/config";
+//import crypto from "crypto";
+
+const projectRouter = Router();
+
+// GET details of specific team using ownerId (staff only)
+projectRouter.get(
+    "/owner/:ownerId/",
+    specification({
+        method: "get",
+        path: "/project/owner/{ownerId}/",
+        tag: Tag.PROJECT,
+        role: Role.STAFF,
+        summary: "Retrieve details of a team using ownerId",
+        parameters: z.object({
+            ownerId: UserIdSchema,
+        }),
+        body: ProjectProjectsSchema,
+        responses: {
+            [StatusCode.SuccessOK]: {
+                description: "The team for this specific ownerId",
+                schema: ProjectProjectsSchema,
+            },
+            [StatusCode.ClientErrorConflict]: {
+                description: "No team found for this ownerId",
+                schema: NoTeamFoundErrorSchema,
+            },
+        },
+    }),
+    async (req, res) => {
+        const ownerId = req.params.ownerId;
+        const projectDetails = await Models.ProjectInfo.findOne({ ownerId });
+
+        // Return no project found if no details were found
+        if (!projectDetails) {
+            return res.status(StatusCode.ClientErrorConflict).send(NoTeamFoundError);
+        }
+
+        return res.status(StatusCode.SuccessOK).send(projectDetails);
+    },
+);
+
+// GET details of user's team
+projectRouter.get(
+    "/details",
+    specification({
+        method: "get",
+        path: "/project/details/",
+        tag: Tag.PROJECT,
+        role: null,
+        summary: "get details of user's team",
+        body: ProjectProjectsSchema,
+        responses: {
+            [StatusCode.SuccessOK]: {
+                description: "The user's team",
+                schema: ProjectProjectsSchema,
+            },
+            [StatusCode.ClientErrorConflict]: {
+                description: "No team found for the user",
+                schema: NoTeamFoundErrorSchema,
+            },
+        },
+    }),
+    async (req, res) => {
+        const { id: userId } = getAuthenticatedUser(req);
+        const userMapping = await Models.ProjectMapping.findOne({ userId });
+        if (!userMapping) {
+            return res.status(StatusCode.ClientErrorNotFound).send(NoTeamFoundError);
+        }
+        const ownerId = userMapping?.teamOwnerId;
+
+        // find project associated with the ownerId
+        const projectDetails = await Models.ProjectInfo.findOne({ ownerId: ownerId });
+
```

Comment:
Return a NoTeamFoundError if projectDetails is undefined. Even though that situation should never happen, we want to be comprehensive of all possible error cases.